pytools/graph.py

```diff
@@ -43,29 +49,46 @@
 Type Variables Used
 -------------------
 
-.. class:: T
+.. class:: NodeT
+
+    Type of a graph node, can be any type.
 
-    Any type.
+.. class:: GraphT
+
+    A :class:`collections.abc.Mapping` representing a directed
+    graph. The mapping contains one key representing each node in the
+    graph, and this key maps to a :class:`collections.abc.Collection` of its
+    successor nodes. Note that most functions expect that every graph node
+    is included as a key in the graph.
 """
 
 from typing import (Collection, TypeVar, Mapping, List, Optional, Any,
-                    Callable, Set, MutableSet, Dict, Iterator, Tuple, FrozenSet)
+                    Callable, Set, MutableSet, Dict, Iterator, Tuple, FrozenSet,
+                    Type)
+
+try:
+    from typing import TypeAlias
+except ImportError:
+    from typing_extensions import TypeAlias
+
+
+NodeT = TypeVar("NodeT")
 
 
-T = TypeVar("T")
+GraphT: TypeAlias = Mapping[NodeT, Collection[NodeT]]
 
 
 # {{{ reverse_graph
 
-def reverse_graph(graph: Mapping[T, Collection[T]]) -> Dict[T, FrozenSet[T]]:
+def reverse_graph(graph: GraphT) -> Dict[NodeT, FrozenSet[NodeT]]:
```

Comment:
```suggestion
def reverse_graph(graph: GraphT) -> GraphT:
```
? Does that work or not because of the `Collection` vs `FrozenSet`?

Reply:
It does work actually, I changed this in [dccdf19](https://github.com/inducer/pytools/pull/166/commits/dccdf19bdf3a6775235fe5aab5d381368e027bd6)